src/app/modules/submission/encrypt-submission/encrypt-submission.middleware.ts

```diff
@@ -87,3 +119,303 @@ export const validateStorageSubmissionParams = celebrate({
+      meta: logMeta,
+      error: formResult.error,
+    })
+    const { errorMessage, statusCode } = mapRouteError(formResult.error)
+    return res.status(statusCode).json({ message: errorMessage })
+  }
+
+  req.body.formsg.formDef = formResult.value
+  return next()
+}
+
+export const checkPublicKey: SharedSubmissionMiddlewareHandlerType = async (
+  req,
+  res,
+  next,
+) => {
+  const formDef = req.body.formsg.formDef
+
+  if (!formDef) return res.send(new FormDefinitionNotRetrievedError())
+
+  const logMeta = {
+    action: 'checkPublicKey',
+    ...createReqMeta(req),
+    formId: formDef.id,
+  }
+
+  // Retrieve public key.
+  const publicKey = formDef.publicKey
+
+  if (!publicKey) {
+    logger.warn({
+      message: 'Form does not have a public key',
+      meta: logMeta,
+    })
+    return res.status(StatusCodes.INTERNAL_SERVER_ERROR).json({
+      message: 'Form does not have a public key',
+    })
+  }
+
+  return next()
+}
+
+export const checkEncryptMode: SharedSubmissionMiddlewareHandlerType = async (
+  req,
+  res,
+  next,
+) => {
+  const formDef = req.body.formsg.formDef
+
+  if (!formDef) return res.send(new FormDefinitionNotRetrievedError())
+
+  const logMeta = {
+    action: 'checkEncryptMode',
+    ...createReqMeta(req),
+    formId: formDef.id,
+  }
+
+  const checkFormIsEncryptModeResult = checkFormIsEncryptMode(formDef)
+  if (checkFormIsEncryptModeResult.isErr()) {
+    logger.error({
+      message:
+        'Trying to submit non-encrypt mode submission on encrypt-form submission endpoint',
+      meta: logMeta,
+    })
+    const { statusCode, errorMessage } = mapRouteError(
+      checkFormIsEncryptModeResult.error,
+    )
+    return res.status(statusCode).json({
+      message: errorMessage,
+    })
+  }
+
+  req.body.formsg.encryptedFormDef = checkFormIsEncryptModeResult.value
+  return next()
+}
+
+/**
+ * Guardrail to prevent new endpoint from being used for regular storage mode forms.
+ * TODO (FRM-1232): remove this guardrail when encryption boundary is shifted.
+ */
+export const checkNewBoundaryEnabled: SharedSubmissionMiddlewareHandlerType =
+  async (req, res, next) => {
+    const formDef = req.body.formsg.formDef
+
+    if (!formDef) return res.send(new FormDefinitionNotRetrievedError())
+
+    if (!formDef.get(newEncryptionBoundaryFlag)) {
+      return res
+        .status(StatusCodes.FORBIDDEN)
+        .json({ message: 'This endpoint has not been enabled for this form.' })
+    }
+
+    return next()
+  }
+
+export const validateSubmission: StorageSubmissionMiddlewareHandlerType =
+  async (req, res, next) => {
+    const formDef = req.body.formsg.formDef
+
+    if (!formDef) return res.send(new FormDefinitionNotRetrievedError())
+
+    const logMeta = {
+      action: 'validateSubmission',
+      ...createReqMeta(req),
+      formId: formDef.id,
+    }
+
+    // Validate submission
+    return await SubmissionService.validateAttachments(req.body.responses)
```

Comment:
Here - file size limits are higher for storage mode

Reply:
Same as the above comment - fixed here: [31847c4](https://github.com/opengovsg/FormSG/pull/6595/commits/31847c41c2edd8514e97779195f46e2a7a239996)